src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -2194,13 +2195,20 @@ public override bool HandleCall(MethodIL callingMethodBody, MethodDesc calledMet
                                 }
 
                                 comDangerousMethod |= IsComInterop(marshalAsDescriptor, calledMethod.Signature[paramIndex]);
+                                abstractDelegatesMethod |= IsAbstractDelegatesInterop(marshalAsDescriptor, calledMethod.Signature[paramIndex]);
                             }
 
                             if (comDangerousMethod)
                             {
                                 reflectionContext.AnalyzingPattern();
                                 reflectionContext.RecordUnrecognizedPattern(2050, $"P/invoke method '{calledMethod.GetDisplayName()}' declares a parameter with COM marshalling. Correctness of COM interop cannot be guaranteed after trimming. Interfaces and interface members might be removed.");
                             }
+
+                            if (abstractDelegatesMethod)
```

Comment:
Do we also need something that warns about fields in marshaled structs?

Reply:
I think I at least should add tests for that, and seems to be I do not nee that for GtkSharp and WPF. Can I do that in the future if this is needed?

Reply:
Hmmm, if we want to do this for structs, maybe a better spot to place all of this detection would be in src\coreclr\tools\aot\ILCompiler.Compiler\Compiler\UsageBasedInteropStubManager.cs, `AddParameterMarshallingDependencies` because that's where we drill into the structs and figure out the dependencies in general. We don't have a logger there, but it's easy to pass one in the `UsageBasedInteropStubManager` constructor.

Reply:
Implement that suggestion